### ISSUE TITLE
NVDA cannot read html link text if the link has an aria-label

### ISSUE THREAD

**Issue by Adriani90** [nvaccess/nvda#16363]
### Steps to reproduce:
1. Take this code example and paste it into the address bar of the browser
```
data:text/html,<div> <body><strong>Useful links</strong><br><a href="blank" download="SwitchAudioFileConverter.exe" aria-label="opens in a new tab">Windows Download</a><br> <a href="../components/switchmaci.zip">Mac Download</a><a href="blank" download="SwitchAudioFileConverter.exe" aria-label="Android download">Android Download</a></body></div>
```
Use case in the wild, this website contains lots of such links, making it almost impossible to work with it accessibly:
https://www.nch.com.au/switch/index.html
2. Try to read all links from the code snippet above in browse and focus mode, by pressing nvda+tab, nvda+shift+o, by using the mouse or by using object navigation.

### Actual behavior:
NVDA reports only aria-label "opens in a new tab" on the link with the link text "Windows download", ignoring the link text. There is no way to read the actual link text unless you use the mouse or object navigation.

### Expected behavior:
1. NVDA should report both link text and aria-label in this order "Windows download, opens in a new tab" on the first link
2. NVDA should filter out the aria-label if aria-label and link text  are the same (e.g. on the "Android download" link). Native HTML semantics should have priority over aria in this case.

### NVDA logs, crash dumps and other attachments:
None
#### NVDA installed/portable/running from source:
Installed

#### NVDA version:
2024.1

#### Windows version:
Windows 11 23 H2

#### Name and version of other software in use when reproducing the issue:
Firefox 124, Chrome and Edge 123.

#### Other information about your system:
n/a

#### Does the issue still occur after restarting your computer?
yes

#### Have you tried any other versions of NVDA? If so, please report their behaviors.
2023.3 has the same issue.

#### If NVDA add-ons are disabled, is your problem still occurring?
yes

#### Does the issue still occur after you run the COM Registration Fixing Tool in NVDA's tools menu?
yes

**Comment by Adriani90** [comment]
I think this has been introduced some versions ago to avoid double speaking in certain cases where aria-label and link text are the same, so it will be either tricky to fix or we will have to reintroduce double speaking and live with that.
Maybe it is possible to let only the link text announced in browse mode, and both of them be announced in focus mode? Then the impact of double speaking would not be that hard since most people use browse mode most of the time.

cc: @jcsteh, @michaelDCurran, @SaschaCowley maybe you have some suggestions on how to proceed, is there a programatic way to detect whether they contain the same text at all?

Aria spec does not explicitely point people to use different content, it even encourages to use the same in aria-label and link text but this is not a requirement as far as I understand.
https://www.w3.org/WAI/WCAG22/Techniques/aria/ARIA8.html#:~:text=In%20some%20assistive%20technologies%20the,text%20supplied%20within%20the%20link.

**Comment by jcsteh** [comment]
It is common understanding (and desire) in the web development community that aria-label should override the content of links, buttons, etc. Unfortunately, this does cause problems in situations of extremely poor authoring like this, but there will always be ways to break things for users due to poor authoring and we can't compensate for all of them without breaking legitimate use cases. It might be reasonable to have some command to report the content of links despite the presence of ARIA labels as a last resort to deal with bad authoring, but I don't think it makes sense to always report this.

**Comment by jcsteh** [comment]
As a side note, I will probably add a feature to kill ARIA labels to my [AxSHammer Firefox add-on](https://addons.mozilla.org/en-US/firefox/addon/axshammer/), which is designed to help deal with poor authoring like this as a last resort. In the interim, you can work around this by pasting this code into the Web Console:

`for (const el of document.querySelectorAll("[aria-label]")) el.ariaLabel = null;`

**Comment by michaelDCurran** [comment]
This is by design. Authors should be using aria-description / aria-describedby for this case. We cannot work around all bad authoring unfortunately. Consider opening a separate issue proposing a new command to specifically read overridden linked text.
